### PR TITLE
Generate cms pages

### PR DESCRIPTION
**Pull request by anxolin**
Follow up on https://github.com/cowprotocol/cow-fi/pull/141

Load CMS data and generate all CMS pages

image

Make uses of the new CMS types, and build a client using `openapi-fetch`, which is the companion library for  `openapi-typescript`

The PR replaces the mock data from previous PRs (and the temporary types for the Article) for the real data and types.

It will implement:
  * `getArticleSlugs` Get all the articles and their slugs. It will fetch all pages (50 items at a time)
  * `getArticles`: Get a list of articles. You can specify the page and the number of items (0 and 50 respectively by default)
  * `getArticleBySlug`: Returen an article given the slug

The previous methods are used to generated all the pages from the CMS at build time.

## URLs

* `/blog`: Entry point for the blog
* `/blog/articles/<slug>`: Article in the blog

## Not included
For now, I don't focus on presenting the information or getting all relevant data. I just present some minimal information about each article

## Test
Run locally, or test the URL

https://cowfi-git-generate-cms-pages-cowswap.vercel.app/blog

### CODE REVIEW

File: pages/blog/articles/[id].tsx [hunk 1]
```diff
@@ -55,7 +57,8 @@ export async function getStaticPaths() {
 }
 
 export const getStaticProps: GetStaticProps<BlogPostProps> = async ({ params }) => {
-  const article = await getArticleBySlug(params.slug as string)
+  console.log('params', params)
```
Comment (alfetopito) [inline pages/blog/articles/[id].tsx:60]:
```suggestion
```

File: pages/blog/articles/[id].tsx [hunk 2]
```diff
@@ -55,7 +57,8 @@ export async function getStaticPaths() {
...
+  const article = await getArticleBySlug(params.id as string)
```
Comment (alfetopito) [inline pages/blog/articles/[id].tsx:61]: Shouldn't params props be typed?

File: services/cms/index.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+export type Article = components["schemas"]["ArticleListResponseDataItem"]
+
+
+/**
+ *
+ * @returns Return all article slugs
+ */
+export async function getArticleSlugs(): Promise<string[]> {
+  // Fetch a single page
+  const fetchPage = async (pageNumber: number) => {
```
Comment (alfetopito) [inline services/cms/index.ts:25]: Nitpick: does it need to be declared inside `getArticleSlugs` fn?

**Review by elena-zh — APPROVED** [review]
Great! But I've created #200 task to iterate on the page's responsiveness

**Comment by anxolin** [comment]
This PR needs to be revisited after https://github.com/cowprotocol/cms/pull/13 and https://github.com/cowprotocol/cms/pull/12 and https://github.com/cowprotocol/cms/pull/11 are merged

File: services/cms/index.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
+import { CmsClient, components } from "@cowprotocol/cms";
```
Comment (anxolin) [inline services/cms/index.ts:1]: @alfetopito I hope you are happy now, u made me suffer to leave this one ready :P

File: pages/blog/articles/[id].tsx (hunk 2 above)
Comment (anxolin) [reply pages/blog/articles/[id].tsx:59]: Okay, will make a private function for it

Comment (anxolin) [reply pages/blog/articles/[id].tsx]: This is how it was done in all the app before :) 
Not meaning that it was good. I've improved it with this commit. cc/ @fairlighteth  https://github.com/cowprotocol/cow-fi/pull/193/commits/5e4e7a3414caf3b5240d44e2dec55cfab056c14a

Comment (anxolin) [reply pages/blog/articles/[id].tsx]: Note that we are also more specific with the param name now: 
image

**Review by alfetopito — APPROVED** [review]
It is working nicely, and barely needed any effort 🤓

**Pull request merged**